crates/sui-types/src/messages.rs

```diff
@@ -3052,6 +3086,30 @@ impl TransactionEffects {
             effects: self.digest(),
         }
     }
+
+    pub fn estimate_effects_size_upperbound(
+        num_writes: usize,
+        num_mutables: usize,
+        num_deletes: usize,
+        num_deps: usize,
+    ) -> usize {
```

Comment:
Given the way this is written, thoughts on just capping `num_writes + num_mutables + num_deletes`?
`num_deps` is already upper bounded by the sum of those three.
It feels a bit strange to basically do math to turn this in bytes, when you could just divide out all of the size information and have a simpler limit. Is there any motivation to actually doing this math? (other than writes are weighted slightly higher?)

Reply:
> num_writes + num_mutables + num_deletes?
num_deps is already upper bounded by the sum of those three.

Yes there's many ways we can structure the formula but without proper weighting, there were lots of overestimations and many existing tests were failing. So I tried to use tighter bounds.

> Is there any motivation to actually doing this math? (other than writes are weighted slightly higher?)

Purely for weighting and consistency purposes. We use number of bytes as our metric for measuring effects impact on the network, so it made sense.

I wouldn't get too worried about these formulas right now. This is an added layer of protection which we should hardly hit because we have internal runtime limits.